### PR TITLE
PR for all completed ouputs

### PR DESCRIPTION
**Pull request by thecodingpoet**
#### What does this PR do?
It shows;
-  Setting up front-end for `writing-html-js-css` 
- InvertedIndex class for `solving-the-problem` output
- Test specs for `writing-tests` output
- Link to a successful TravisCI build for`setting-up-continuous-integration` output
- Link to medium post for `reflecting-on-the-learning-experience` output
- Link to herokuapp pages for `deployment` output
- Link to release notes for `delivering-your-product` output
- Two design patterns for `using-design-patterns` output

#### Description of Task to be completed?
- A Pull Request showing your implementation of the solution (e.g. InvertedIndex class)
- A link to a successful TravisCI build of your `development` branch (it should include test runs)
- A folder containing the template for the proposed front-end of your project
- A Pull Request showing all tests required for your project using Mocha/Jasmine with any assertion library of choice (Chai.JS, Should.JS, Expect.JS...)
- A post reflecting on your learning experience while producing these outputs
- Live URLs to a `review` app, `staging` & `production` apps
- `Release notes` for your final product
- A Pull Request showing two Design Patterns (suggestions: Decorator, Factory, Observer, Prototype...)

#### How should this be manually tested?

- For `writing-html-js-css` output: Check Screenshots below:
- For `solving-the-problem` output: View [Solving the problem](https://github.com/andela/test-simulations-mgmt-repo/tree/bayo-izanagi-oduyile/06-building-a-program-start-to-finish/solving-the-problem/output) output folder
- For `writing-tests` output: Visit [Test output folder](https://github.com/andela/test-simulations-mgmt-repo/tree/bayo-izanagi-oduyile/06-building-a-program-start-to-finish/writing-tests/output/)
- For`setting-up-continuous-integration` output: [Travis on Development](https://travis-ci.org/andela-oduyile/Inverted-Index-App)
- For `reflecting-on-the-learning-experience` output: [Medium post](https://medium.com/@oluwatomi.duyile/the-journey-to-become-a-world-class-developer-27d905986899#.7v9ffg1ch")
- For `deployment` output: Click the links below
 1. [Review App](https://indexit-pr-10.herokuapp.com/) 2. [Staging App](https://indexit-staging.herokuapp.com/) 3. [Production App](https://indexit.herokuapp.com/)

- For `delivering-your-product` output: [Release Document](https://docs.google.com/document/d/1LCc83vH6kCfGZIB1vxNpqT_51nWHoVzJB6L2kQdicAs/edit)
- For `using-design-patterns` output: [Test output folder](https://github.com/andela/test-simulations-mgmt-repo/tree/bayo-izanagi-oduyile/06-building-a-program-start-to-finish/using-design-patterns/output)

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
N/A

#### Screenshots (if appropriate)

screen shot 2017-03-16 at 9 32 18 pm
screen shot 2017-03-16 at 8 00 29 pm
screen shot 2017-03-17 at 1 42 06 am
screen shot 2017-03-17 at 1 43 23 am
screen shot 2017-03-17 at 1 45 44 am
screen shot 2017-03-17 at 4 28 27 am

### CODE REVIEW

File: 06-building-a-program-start-to-finish/delivering-your-product/output/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#### Output
+- **Using**: Google Doc
+- **Deliver**: `Release notes` for your final product
+- **With**: *independently*
+
+[Release Note](https://docs.google.com/document/d/1LCc83vH6kCfGZIB1vxNpqT_51nWHoVzJB6L2kQdicAs/edit)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/delivering-your-product/output/README.md:6]: You have a typo in your document. Other than that good work. However please correct it.

File: 06-building-a-program-start-to-finish/checking-repository-health/output/README.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#### Output
+- **Using**: `Code Climate` and `Coveralls`
+- **Deliver**: a screenshot of your code coverage and GPA badges in your repo's README file
+- **With**: *independently*
+
+![image](https://cloud.githubusercontent.com/assets/25608336/24113866/76842f96-0d9e-11e7-8366-b66462e2ad9d.png)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/checking-repository-health/output/README.md:6]: Nice

File: 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/output/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#### Output
+- **Using**: Medium
+- **Deliver**: a post reflecting on your learning experience while producing these outputs
+- **With**: *independently*
+
+#### Targeted Learning Outcomes
+[Go to medium post](https://medium.com/@oluwatomi.duyile/the-journey-to-become-a-world-class-developer-27d905986899#.77u1ap5sa)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/output/README.md:7]: Nice read

File: 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/output/README.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#### Output
+- **Using**: TravisCI
+- **Deliver**: a link to a successful TravisCI build of your `development` branch (it should include test runs)
+- **With**: *independently*
+
+[Successful Travis Build](https://travis-ci.org/andela-oduyile/Inverted-Index-App)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/output/README.md:6]: Nice

File: 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/output/README.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#### Output
+live URLs to a `review` app, your `staging` & `production` apps
+- **With**: *independently*
+
+- [Link to review app](https://indexit-pr-10.herokuapp.com/)
+- [Link to staging app](https://indexit-staging.herokuapp.com/)
+- [Link to production app](https://indexit.herokuapp.com/)
+
+#### AUTHOR
+- **Duyile Oluwatomi**
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/output/README.md:10]: Any reason why your "review app" link is not working?

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js [hunk 6]
```diff
@@ -0,0 +1,276 @@
+/* eslint-disable no-unused-vars */
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: I believe you can have all your major functions in InvertedIndex.js. Just seems like the invertedIndex class is doing little. Please address.

File: 06-building-a-program-start-to-finish/using-design-patterns/output/README.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+```
+const invertedIndex = new InvertedIndex();
+
+invertedIndex.createIndex('books.json', fileContent);
+
+invertedIndex.getIndex('books.json');
+
+invertedIndex.searchIndex(query, 'books.json');
+
+```
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/using-design-patterns/output/README.md:52]: Ok

**Review by andela-moe — COMMENTED** [review]
Please address comments

File: 06-building-a-program-start-to-finish/writing-tests/output/tests.js [hunk 8]
```diff
@@ -0,0 +1,209 @@
+
+/* eslint-disable no-undef */
+
+const data = require('./data.json');
+const book = require('./books.json');
+const invalid = require('./invalid.json');
+const invalidContent = require('./invalid-content.json');
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/tests.js]: A base filename should exactly match the name of its default export. [See link](https://github.com/airbnb/javascript#naming--filename-matches-export)

File: 06-building-a-program-start-to-finish/writing-tests/output/tests.js [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+        ['is', true, true],
+        ['magic', false, true]
+      ];
+      expect(actual).toEqual(expected);
+    });
+    it('should get search result for all files indexded', () => {
+      const query = 'alice is lost in the ring';
+      inverted.createIndex('data.json', data);
+      inverted.createIndex('books.json', book);
+      const actual = inverted.searchIndex(query, 'All').index;
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/tests.js]: "actual" is defined but never used.

File: 06-building-a-program-start-to-finish/writing-tests/output/tests.js [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+      const actual = inverted.searchIndex(query, 'data.json').index;
+      const expected = [
+        ['harry', false, true],
+        ['lost', true, true],
+        ['is', true, true],
+        ['magic', false, true]
+      ];
+      expect(actual).toEqual(expected);
+    });
+    it('should get search result for all files indexded', () => {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/tests.js]: Is there an expected outcome for this test?

**Review by andela-moe — COMMENTED** [review]
Please address comments

File: 06-building-a-program-start-to-finish/writing-tests/output/tests.js [hunk 11]
```diff
@@ -0,0 +1,209 @@
+
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/tests.js]: Please follow the proper naming convention for test files.

File: 06-building-a-program-start-to-finish/delivering-your-product/output/README.md (hunk 1 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/delivering-your-product/output/README.md:6]: Done

File: 06-building-a-program-start-to-finish/writing-tests/output/tests.js (hunk 9 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/writing-tests/output/tests.js]: Done

File: 06-building-a-program-start-to-finish/writing-tests/output/tests.js (hunk 10 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/writing-tests/output/tests.js]: Yes, there is. It checks if the result gotten from the search index method is equal to the expected variable. I had to create index before i search  though.

File: 06-building-a-program-start-to-finish/writing-tests/output/tests.js (hunk 8 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/writing-tests/output/tests.js]: Done

File: 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/output/README.md (hunk 5 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/output/README.md:10]: Fixed this

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js (hunk 6 above)
Comment (andela-oduyile) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: You said you would get back to me on this.

File: 06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js [hunk 12]
```diff
@@ -0,0 +1,88 @@
+/* eslint-disable no-undef */
+/* eslint-disable no-unused-vars */
+
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js]: Why are these here please?

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js [hunk 13]
```diff
@@ -0,0 +1,276 @@
...
+/* eslint-disable no-undef */
+
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: Why are these here please?

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js [hunk 14]
```diff
@@ -0,0 +1,276 @@
...
+/**
+   * filterIndexed method takes an already indexed array
+   * and filters it to match the query
+   * @param {Array} query - An array of the words
+   * gotten from the sanitized query
+   * @param {Array} indexedFile - An already indexed file
+   * @return {Array} - The queried indexed array
+   * @memberOf InvertedIndexUtility
+   */
+  static filterIndexed(query, indexedFile) {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: what is "query"? is that supposed to be a parameter name? I believe you can used something that best describes what the parameter is.

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js [hunk 15]
```diff
@@ -0,0 +1,276 @@
...
+
+  /**
+   * sanitizeQuery method removes unwanted terms in the query string
+   * and also converts the query string into an array of each unique term
+   * @static
+   * @param {String} query - The terms to search for the index
+   * @returns {Array} - The already formatted query
+   * @memberOf InvertedIndexUtility
+   */
+  static sanitizeQuery(query) {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: Your method and parameter names are inappropriate. Please names that best describes them.

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js [hunk 16]
```diff
@@ -0,0 +1,276 @@
...
+  static constructIndex(words, wordsInEachBook, bookTitles) {
+    const checked = this
+    .checkOccurenceOfWords(words, wordsInEachBook);
+    const booksLength = bookTitles.length;
+    const booksChecked = this
+    .getWordOccurenceForEachBook(checked, booksLength);
+    const indexConstructed = this
+    .mapWordsWithOccurence(words, booksChecked);
+    return indexConstructed;
+  }
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: Your method names are too long.. Do you think you can make them shorter and still communicate their purpose?

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js [hunk 17]
```diff
@@ -0,0 +1,276 @@
...
+   * @returns {Array} - An array that consist of
+   * booleans to indicate if the words are present in each book
+   * e.g [true, false, false, false, false, true . . .]
+   * @memberOf InvertedIndexUtility
+   */
+  static checkOccurenceOfWords(words, wordsInEachBook) {
+    const checked = [];
+
+    words.forEach((word) => {
+      for (let i = 0; i < wordsInEachBook.length; i += 1) {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: This right here is a violation of the Airbnb Style Guide. See [Link](https://github.com/airbnb/javascript#iterators--nope)

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js [hunk 18]
```diff
@@ -0,0 +1,276 @@
...
+
+  /**
+   * @param {Array} file - contains an array of objects that
+   * contain each book with their title and text.
+   * @returns {Array} - an Array that contains all the words in
+   * the book without duplication
+   */
+  static getAllWords(file) {
+    let words = '';
+    for (let i = 0; i < file.length; i += 1) {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: This right here is a violation of the Airbnb Style Guide. See [Link](https://github.com/airbnb/javascript#iterators--nope)

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js [hunk 19]
```diff
@@ -0,0 +1,276 @@
...
+  /**
+   * This method gets the title of each book in a file
+   * @static
+   * @param {Array} file - The content of the given file
+   * @returns {Array} - The Title of each book in the file
+   * @memberOf InvertedIndexUtility
+   */
+  static getTitlesOfEachBook(file) {
+    const titleList = [];
+    for (let i = 0; i < file.length; i += 1) {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: This right here is a violation of the Airbnb Style Guide. See [Link](https://github.com/airbnb/javascript#iterators--nope)

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js [hunk 20]
```diff
@@ -0,0 +1,276 @@
...
+   * @static
+   * @param {Array} file - contains an array of objects that
+   * contain each book with their title and text.
+   * @returns {Array} - an array that consist of all the words in each book
+   * @memberOf InvertedIndexUtility
+   */
+  static getAllWordsInEachBook(file) {
+    const doc = [];
+    let bookContent;
+    for (let i = 0; i < file.length; i += 1) {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: This right here is a violation of the Airbnb Style Guide. See [Link](https://github.com/airbnb/javascript#iterators--nope)

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js [hunk 21]
```diff
@@ -0,0 +1,276 @@
...
+   * @param {Integer} bookCount - The number of the book present in the file
+   * @returns {Array} - An Array that consist of sub arrays of each word
+   * and a boolean to indicate their presence in each book
+   * @memberOf InvertedIndexUtility
+   */
+  static getWordOccurenceForEachBook(checked, bookCount) {
+    const result = [];
+    const stopIndex = Math.ceil(checked.length / bookCount);
+
+    for (let i = 0; i <= stopIndex; i += 1) {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: This right here is a violation of the Airbnb Style Guide. See [Link](https://github.com/airbnb/javascript#iterators--nope)

Comment (andela-moe) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: Please fix this all across this file

**Review by andela-moe — COMMENTED** [review]
Please address comments

File: 06-building-a-program-start-to-finish/writing-tests/output/tests.js (hunk 11 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/writing-tests/output/tests.js]: Done

File: 06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js (hunk 12 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js]: i am calling some static methods in my InvertedIndexUtility class and es-lint is saying InvertedIndexUtility is not defined. So i had to ignore it.

Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js]: Also creating a class without exporting flags a violation. So i ignored that too

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js (hunk 13 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: Same as above

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js (hunk 14 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: Done - changed query to searchTerms

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js (hunk 19 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: Done

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js (hunk 18 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: Done

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js (hunk 20 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: Done

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js (hunk 17 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: Done

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js (hunk 20 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: Done

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js (hunk 18 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: Done

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js (hunk 21 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: All Done

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js (hunk 15 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: Fixed this

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js (hunk 16 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: I have tried shortening the ones i can

File: 06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js [hunk 22]
```diff
@@ -68,7 +67,7 @@ class InvertedIndex {
         const fileIndexed = this.getIndex(name);
         bookNames.push(fileIndexed.title);
         result.push(InvertedIndexUtility
-        .filterIndexed(query, fileIndexed.index));
+        .filterIndexed(searchTerms, fileIndexed.index));
       });
       searchResult = (fileNames.length !== 0) ?
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js]: Please confirm this is the best way of using ternary operator. [See link.](https://github.com/airbnb/javascript#comparison--unneeded-ternary)

File: 06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js [hunk 23]
```diff
@@ -77,7 +76,7 @@ class InvertedIndex {
       const fileIndexed = this.getIndex(fileName);
       const bookNames = fileIndexed.title;
       const result = InvertedIndexUtility
-      .filterIndexed(query, fileIndexed.index);
+      .filterIndexed(searchTerms, fileIndexed.index);
       searchResult = fileIndexed ?
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js]: Please confirm this is the best way of using ternary operator. [See link.](https://github.com/airbnb/javascript#comparison--unneeded-ternary)

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js [hunk 24]
```diff
@@ -0,0 +1,254 @@
...
+   * This function returns an array that consists of all the words in each book
+   * [[], [], ]
+   * @static
+   * @param {Array} file - contains an array of objects that
+   * contain each book with their title and text.
+   * @returns {Array} - an array that consist of all the words in each book
+   * @memberOf InvertedIndexUtility
+   */
+  static getAllWordsInEachBook(file) {
+    const doc = [];
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: Is "doc" a word or an abbreviation? Can this be written in full such that it communicates correctly?

File: 06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js [hunk 25]
```diff
@@ -0,0 +1,87 @@
+/* eslint-disable no-undef */
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js]: Is "inverted-index.js" the appropriate naming convention for this file's name?

**Review by andela-moe — COMMENTED** [review]
Please address comments

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js (hunk 24 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: Done

File: 06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js (hunk 25 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js]: Fixed this

File: 06-building-a-program-start-to-finish/writing-tests/output/inverted-index-test.js [hunk 26]
```diff
@@ -0,0 +1,201 @@
+/* eslint-disable no-undef */
+
+const data = require('./data');
+const books = require('./books');
+const invalid = require('./invalid');
+const invalidContent = require('./invalidContent');
+
+const inverted = new InvertedIndex();
+
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/inverted-index-test.js:9]: How are you able to use your InvertedIndexUtility class in your test?

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js [hunk 27]
```diff
@@ -0,0 +1,255 @@
...
+   * of all the words in each book
+   * @returns {Array} - An array consisiting of a sub array with each word
+   * and booleans to indicate their occurence in each book
+   * @memberOf InvertedIndexUtility
+   */
+  static constructIndex(words, wordsInEachBook, bookTitles) {
+    const checked = this.checkOccurence(words, wordsInEachBook);
+    const booksLength = bookTitles.length;
+    const booksChecked = this.getEachBookOccurence(checked, booksLength);
+    const indexConstructed = this.mapWordsWithOccurence(words, booksChecked);
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: Your function/method names in your InvertedIndex and InvertedIndexUtility are too long.. though very descriptive.. Please make them shorter and communicate what they actually do.

Remember your JsDoc is there to provide description on what that method/function is doing.

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js [hunk 28]
```diff
@@ -0,0 +1,255 @@
+/* eslint-disable no-unused-vars */
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: The name of this file should be in PascalCase

File: 06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js (hunk 25 above)
Comment (andela-moe) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js]: The name of this file should be in PascalCase

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js [hunk 29]
```diff
@@ -0,0 +1,255 @@
...
+   * @static
+   * @param {Array} checked - This array consist of
+   * booleans to indicate if the words are present in each book
+   * e.g [true, false, false, false, false, true . . .]
+   * @param {Integer} bookCount - The number of the book present in the file
+   * @returns {Array} - An Array that consist of sub arrays of each word
+   * and a boolean to indicate their presence in each book
+   * @memberOf InvertedIndexUtility
+   */
+  static getEachBookOccurence(checked, bookCount) {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: It is usually advised that your parameter names are nouns. "**checked**" is not a noun. Please use an appropriate name for this parameter.

Please address a reoccurrence of this across your other functions/methods

File: 06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js [hunk 30]
```diff
@@ -0,0 +1,87 @@
...
+  }
+
+  /**
+   * getIndex method returns the indexed words and the documents that were found
+   * @param {String} fileName - The name of the file to get its index
+   * @return {Object} - The index already created
+  */
+  getIndex(fileName) {
+    return this.filesIndexed[fileName] ? this.filesIndexed[fileName] : false;
+  }
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js]: What datatype are you returning here?

File: 06-building-a-program-start-to-finish/writing-tests/output/inverted-index-test.js [hunk 31]
```diff
@@ -0,0 +1,201 @@
+/* eslint-disable no-undef */
+
+const data = require('./data');
+const books = require('./books');
+const invalid = require('./invalid');
+const invalidContent = require('./invalidContent');
+
+const inverted = new InvertedIndex();
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/inverted-index-test.js:8]: Can you name this variable after the class that you are instantiating?

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js [hunk 32]
```diff
@@ -0,0 +1,255 @@
...
+      return false;
+    }
+    file.forEach((book) => {
+      if ((!book.title) || (!book.text)) {
+        isValid = false;
+      }
+    });
+    return isValid;
+  }
+}
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: So when a valid **_json_** file without the appropriate key value pairs or an array is uploaded what happens?

**Review by andela-moe — COMMENTED** [review]
Please address comments.

File: 06-building-a-program-start-to-finish/writing-tests/output/inverted-index-test.js (hunk 31 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/writing-tests/output/inverted-index-test.js:8]: Done

File: 06-building-a-program-start-to-finish/writing-tests/output/inverted-index-test.js (hunk 26 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/writing-tests/output/inverted-index-test.js:9]: I included the file in my karma config

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js (hunk 28 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: Done

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js (hunk 27 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: Done

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js (hunk 29 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: Done

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js (hunk 32 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js]: I have another method called isValidContent which checks if the content of the JSON file is valid.

File: 06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js (hunk 30 above)
Comment (thecodingpoet) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js]: I am returning an Object

**Comment by andela-moe** [comment]
Good work
